redis/src/main/scala/zio/redis/RedisEnvironment.scala

```diff
@@ -16,9 +16,12 @@
 
 package zio.redis
 
+import zio.schema.Schema
 import zio.schema.codec.BinaryCodec
 
 private[redis] trait RedisEnvironment {
-  protected def codec: BinaryCodec
+  protected def codec: CodecSupplier
   protected def executor: RedisExecutor
+
+  implicit def summonCodec[A: Schema]: BinaryCodec[A] = codec.codec
```

Comment:
Is it necessary?

Reply:
yeah, we have to "summon" the implicit codec, without this you get like 600 error when trying to compile, is there a better way you think?